applications/ConvectionDiffusionApplication/custom_elements/symbolic_d_convection_diffusion_explicit.h

```diff
@@ -0,0 +1,317 @@
+        typedef Geometry<NodeType>::PointsArrayType NodesArrayType;
+        typedef Vector VectorType;
+        typedef Matrix MatrixType;
+        typedef std::size_t IndexType;
+        typedef std::vector<std::size_t> EquationIdVectorType;
+        typedef std::vector< Dof<double>::Pointer > DofsVectorType;
+        typedef GeometryData::IntegrationMethod IntegrationMethod;
+
+    /// Pointer definition of SymbolicDConvectionDiffusionExplicit
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(SymbolicDConvectionDiffusionExplicit);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    //Constructors.
+
+    /// Default constuctor.
+    SymbolicDConvectionDiffusionExplicit(IndexType NewId, GeometryType::Pointer pGeometry);
+    SymbolicDConvectionDiffusionExplicit(IndexType NewId, GeometryType::Pointer pGeometry, Properties::Pointer pProperties);
+
+    /// Destructor.
+    virtual ~SymbolicDConvectionDiffusionExplicit();
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    Element::Pointer Create(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        Properties::Pointer pProperties) const override;
+
+    Element::Pointer Create(
+        IndexType NewId,
+        GeometryType::Pointer pGeom,
+        Properties::Pointer pProperties) const override;
+
+    void CalculateLocalSystem(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLeftHandSide(
+        MatrixType& rLeftHandSideMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateRightHandSide(
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void AddExplicitContribution(
+        const ProcessInfo &rCurrentProcessInfo) override;
+
+    void Initialize(
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void FinalizeSolutionStep(
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void Calculate(
+        const Variable<double>& rVariable,
+        double& Output,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateOrthogonalSubgridScaleSystem(
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo);
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    int Check(const ProcessInfo &rCurrentProcessInfo) override;
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "SymbolicDConvectionDiffusionExplicitElement #";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << Info() << this->Id();
+    }
+
+    ///@}
+
+protected:
+
+    ///@name Protected member Variables
+    ///@{
+
+    VectorType mUnknownSubScale;
```

Comment:
Couldn't you template this with the `TNumNodes` to be a bounded array?

Reply:
Yes. I have also done [this change](https://github.com/KratosMultiphysics/Kratos/commit/ea0316a929cd6b0dd24c45ff4805300fe837fcc4#diff-4c7c037ea1ddd42d58f38cc37c3ac907L99), since resizing is no more needed.